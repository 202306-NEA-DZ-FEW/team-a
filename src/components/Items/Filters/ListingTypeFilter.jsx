import { Menu, Transition } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { IoMdArrowDropdown } from "react-icons/io";
import { TbSitemap } from "react-icons/tb";

function ListingTypeFilter({ t, queryParams }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  const listingTypes = ["exchangeButton", "requestButton", "donationButton"];

  return (
    <>
      <Menu as='div' className='relative w-full flex justify-center text-left'>
        <Menu.Button className='flex items-center gap-2 w-full justify-between bg-base-200 input-sm rounded-full tracking-wider'>
          <div className='flex gap-2 items-center'>
            <TbSitemap className='w-5 h-5' />
            <span>
              {queryParams.listingType
                ? t(`addItem:${queryParams.listingType}`)
                : t("addItem:listingType")}
            </span>
          </div>
          <IoMdArrowDropdown />
        </Menu.Button>

        <Transition
          as={Fragment}
          enter='transition ease-out duration-100'
          enterFrom='transform opacity-0 scale-95'
          enterTo='transform opacity-100 scale-100'
          leave='transition ease-in duration-75'
          leaveFrom='transform opacity-100 scale-100'
          leaveTo='transform opacity-0 scale-95'
        >
          <Menu.Items className='absolute max-h-52 overflow-y-scroll p-2 no-scrollbar bg-base-100 z-10 rounded-lg top-10 w-full ring-1 ring-base-300 ring-opacity-5 focus:outline-none drop-shadow-xl'>
            <div className='py-1'>
              {listingTypes.map((type) => (
                <Link
                  scroll={false}
                  key={type}
                  href={{
                    pathname: "/items",
                    query: { ...queryParams, listingType: type },
                  }}
                >
                  <Menu.Item>
                    {({ active }) => (
                      <p
                        className={classNames(
                          active ? "bg-base-200" : "text-gray-500",
                          "block px-4 py-2 tracking-wider my-2 rounded-md"
                        )}
                      >
                        {t(`addItem:${type}`)}
                      </p>
                    )}
                  </Menu.Item>
                </Link>
              ))}
            </div>
          </Menu.Items>
        </Transition>
      </Menu>
    </>
  );
}

export default ListingTypeFilter;
