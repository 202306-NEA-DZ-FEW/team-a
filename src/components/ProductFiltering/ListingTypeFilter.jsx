import { Menu, Transition } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { IoMdArrowDropdown } from "react-icons/io";
import { TbSitemap } from "react-icons/tb";

function ListingTypeFilter({ t, queryParams }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  const listingTypes = ["exchangeButton", "requestButton", "donationButton"];

  return (
    <>
      <Menu as='div' className='relative w-full flex justify-center text-left'>
        <Menu.Button className='flex items-center gap-2 w-full justify-between border bg-white input-sm rounded-full tracking-wider'>
          <div className='flex gap-2 items-center'>
            <TbSitemap className='w-5 h-5' />
            <span>
              {queryParams.listingType
                ? t(`addItem:${queryParams.listingType}`)
                : "Listing type"}
            </span>
          </div>
          <IoMdArrowDropdown />
        </Menu.Button>

        <Transition
          as={Fragment}
          enter='transition ease-out duration-100'
          enterFrom='transform opacity-0 scale-95'
          enterTo='transform opacity-100 scale-100'
          leave='transition ease-in duration-75'
          leaveFrom='transform opacity-100 scale-100'
          leaveTo='transform opacity-0 scale-95'
        >
          <Menu.Items className='absolute max-h-52 overflow-y-scroll no-scrollbar bg-white z-10 rounded-lg top-10 w-full ring-1 ring-black ring-opacity-5 drop-shadow-xl'>
            <div className='py-1'>
              {listingTypes.map((type) => (
                <Link
                  scroll={false}
                  key={type}
                  href={{
                    pathname: "/products",
                    query: { ...queryParams, listingType: type },
                  }}
                >
                  <Menu.Item>
                    {({ active }) => (
                      <p
                        className={classNames(
                          active
                            ? "bg-gray-100 text-gray-900"
                            : "text-gray-700",
                          "block px-4 py-2 tracking-wider my-2"
                        )}
                      >
                        {t(`addItem:${type}`)}
                      </p>
                    )}
                  </Menu.Item>
                </Link>
              ))}
            </div>
          </Menu.Items>
        </Transition>
      </Menu>
    </>
  );
}

export default ListingTypeFilter;
