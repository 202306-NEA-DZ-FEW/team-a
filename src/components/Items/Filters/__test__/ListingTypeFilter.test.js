import renderer from "react-test-renderer";

import ListingTypeFilter from "../ListingTypeFilter";

it("renders correctly", () => {
  const queryParams = {};
  const mockT = jest.fn();
  const tree = renderer
    .create(<ListingTypeFilter queryParams={queryParams} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
