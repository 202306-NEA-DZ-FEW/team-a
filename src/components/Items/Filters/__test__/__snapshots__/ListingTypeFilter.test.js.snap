// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative w-full flex justify-center text-left"
  data-headlessui-state=""
>
  <button
    aria-expanded={false}
    aria-haspopup="menu"
    className="flex items-center gap-2 w-full justify-between border bg-white input-sm rounded-full tracking-wider"
    data-headlessui-state=""
    id="headlessui-menu-button-:r0:"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <div
      className="flex gap-2 items-center"
    >
      <svg
        className="w-5 h-5"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M3 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z"
        />
        <path
          d="M15 15m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z"
        />
        <path
          d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z"
        />
        <path
          d="M6 15v-1a2 2 0 0 1 2 -2h8a2 2 0 0 1 2 2v1"
        />
        <path
          d="M12 9l0 3"
        />
      </svg>
      <span />
    </div>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128 192l128 128 128-128z"
      />
    </svg>
  </button>
</div>
`;
